mindsdb_native/libs/backends/lightwood.py

```diff
@@ -278,38 +278,95 @@ def train(self):
-                lightwood_config['mixer']['kwargs']['stop_training_after_seconds'] = self.transaction.lmd['stop_training_in_x_seconds']
+        for mixer_class in mixer_classes:
+            lightwood_config['mixer']['kwargs'] = {}
+            lightwood_config['mixer']['class'] = mixer_class
 
-        logging.getLogger().setLevel(logging.DEBUG)
+            if lightwood_config['mixer']['class'] == lightwood.mixers.NnMixer:
+                # Evaluate less often for larger datasets and vice-versa
+                eval_every_x_epochs = int(round(1 * pow(10, 6) * (1 / len(train_df))))
 
-        self.predictor.learn(from_data=train_df, test_data=test_df)
+                # Within some limits
+                if eval_every_x_epochs > 200:
+                    eval_every_x_epochs = 200
+                if eval_every_x_epochs < 3:
+                    eval_every_x_epochs = 3
 
-        self.transaction.log.info('Training accuracy of: {}'.format(self.predictor.train_accuracy))
+                kwargs = lightwood_config['mixer']['kwargs']
+
+                kwargs['callback_on_iter'] = self.callback_on_iter
+                kwargs['eval_every_x_epochs'] = eval_every_x_epochs
+                kwargs['stop_training_after_seconds'] = self.transaction.lmd['stop_training_in_x_seconds']
+
+            self.predictor = lightwood.Predictor(lightwood_config.copy())
+
+            self.predictor.learn(
+                from_data=lightwood_train_ds,
+                test_data=lightwood_test_ds
+            )
+
+            self.transaction.log.info('[{}] Training accuracy of: {}'.format(
+                mixer_class.__name__,
+                self.predictor.train_accuracy
+            ))
+
+            validation_predictions = self.predict('validate')
+            validation_accuracy = evaluate_accuracy(
+                validation_predictions,
+                self.transaction.input_data.validation_df,
+                self.transaction.lmd['stats_v2'],
+                self.transaction.lmd['predict_columns'],
+                backend=self
+            )
+
+            predictors_and_accuracies.append((
+                self.predictor,
+                validation_accuracy
+            ))
+
+        best_predictor, best_accuracy = max(predictors_and_accuracies, key=lambda x: x[1])
+
+        # Find predictor with NnMixer
+        for predictor, accuracy in predictors_and_accuracies:
+            if isinstance(predictor._mixer, lightwood.mixers.NnMixer):
+                nn_mixer_predictor, nn_mixer_predictor_accuracy = predictor, accuracy
+                break
+        else:
+            nn_mixer_predictor, nn_mixer_predictor_accuracy = None, None
+
+        self.predictor = best_predictor
+
+        # If difference between accuracies of best predictor and NnMixer predictor
+        # is small, then use NnMixer predictor
+        if nn_mixer_predictor is not None:
+            SMALL_ACCURACY_DIFFERENCE = 0.05
```

Comment:
Make 0.01, a difference of 5% seems like it could be significant, 1% might well be noise. 

But, obviously, just a heuristic, still, our focus is on accuracy so I'd like to prioritize that for now.